emcc.py

```diff
@@ -2791,6 +2791,58 @@ def in_directory(root, child):
   return False
 
 
+def parse_string_value(text):
+  first = text[0]
+  if first == "'" or first == '"':
+    assert text[-1] == text[0] and len(text) > 1, 'unclosed opened quoted string. expected final character to be "%s" and length to be greater than 1 in "%s"' % (text[0],text)
+    return text[1:-1]
+  else:
+    return text
+
+def _parse_string_list_inner(text, sep): # Part of parse_value
+  values = text.split(sep)
+  result = []
+  index = 0
+  while True:
+    current = values[index].lstrip() # Cannot remove all at start/end because that could affect a string with a comma
+    first = current[0]
+    if not(first == "'" or first == '"'): # Longer logic in else.
+      assert len(current) > 0, "string array should not contain an empty value"
+      result.append(current.rstrip())
+    else:
+      while True: # Continue until we find a section ending in the quote
+        start = index
+        assert index < len(values), "unclosed quoted string. expected final character to be '%s' in '%s'" % (first, values[start])
+        if values[index].rstrip()[-1] == first: # Final character exlcuding whitespace
+          if start == index:
```

Comment:
hmm, how can `start != index`? perhaps `start = index` should be outside the loop?

Reply:
I did mean for `start = index` to be outside the loop. None of the tests caught this because none include strings containing a comma, and on the first loop it is equivalent.